packages/dashboard/src/components/userPropertiesTable.tsx

```diff
@@ -0,0 +1,222 @@
+        ".MuiDataGrid-cell:focus": {
+          outline: "none",
+        },
+        // pointer cursor on ALL rows
+        "& .MuiDataGrid-row:hover": {
+          cursor: "pointer",
+        },
+      }}
+      getRowId={(row) => row.id}
+      onRowClick={(params) => {
+        router.push({
+          pathname: `/user-properties/${params.id}`,
+        });
+      }}
+      autoPageSize
+      columns={[
+        {
+          field: "properties",
+          headerName: "Name",
+        },
+        {
+          field: "lastRecomputed",
+          headerName: "Last Re-Computed",
+        },
+        {
+          field: "updatedAt",
+          headerName: "Last Updated",
+        },
+        {
+          field: "templates",
+          headerName: "Templates used by",
+          renderCell: ({ row }: { row: Row }) => {
+            const currentRow = row;
+            if (currentRow.templates.length === 0) {
+              return (
+                <div>
+                  <p>No Templates</p>
+                </div>
+              );
+            }
+            return (
+              <div>
+                {currentRow.templates.map((template) => {
+                  let type = "email";
+                  if (template.type === "Email") {
+                    type = "email";
+                  } else if (template.type === "Sms") {
+                    type = "sms";
+                  } else if (template.type === "MobilePush") {
+                    type = "mobile-push";
+                  }
+                  return (
+                    <Tooltip title={template.name} key={template.id}>
+                      <Link
+                        href={`/templates/${type}/${template.id}`}
+                        passHref
+                        onClick={(e) => {
+                          e.stopPropagation();
+                        }}
+                        style={{
+                          display: "block",
+                          margin: "0.2rem 0",
+                          backgroundColor: "#f5f5f5",
+                          padding: "0.5rem",
+                          borderRadius: "0.5rem",
+                          overflow: "hidden",
+                          textOverflow: "ellipsis",
+                          whiteSpace: "nowrap",
+                          textDecoration: "underline",
+                          width: 200,
+                          maxWidth: "fit-content",
+                        }}
+                      >
+                        {template.name}
+                      </Link>
+                    </Tooltip>
+                  );
+                })}
+              </div>
+            );
+          },
+        },
+        {
+          field: "actions",
+          headerName: "Action",
+          width: 180,
+          sortable: false,
+          // eslint-disable-next-line react/no-unused-prop-types
+          renderCell: ({ row }: { row: Row }) => {
+            const onClick = () => {
```

Comment:
Why are you initializing a function only to immediately call it here and in other parts of the code?

Reply:
I have made the changes now. Please have a look.